packages/hub/node-tests/tasks/send-email-card-drop-verification-test.ts

```diff
@@ -0,0 +1,134 @@
+describe('SendEmailCardDropVerificationTask', function () {
+  this.beforeAll(function () {
+    registry(this).register('email', StubEmail);
+  });
+
+  let { getContainer } = setupHub(this);
+  this.beforeEach(async function () {
+    StubEmail.shouldThrow = false;
+    StubEmail.lastSent = null;
+    Sentry.init({
+      dsn: 'https://acacaeaccacacacabcaacdacdacadaca@sentry.io/000001',
+      release: 'test',
+      tracesSampleRate: 1,
+      transport: sentryTransport,
+    });
+    let cardDropQueries = (await getContainer().lookup('email-card-drop-requests', {
+      type: 'query',
+    })) as EmailCardDropRequestsQueries;
+    await cardDropQueries.insert(cardDropRequest);
+  });
+
+  it('sends an email that contains the verification link', async function () {
+    let task = (await getContainer().lookup('send-email-card-drop-verification')) as SendEmailCardDropVerification;
+    let params = new URLSearchParams();
+    params.append('eoa', cardDropRequest.ownerAddress);
+    params.append('verification-code', cardDropRequest.verificationCode);
+    params.append('email-hash', cardDropRequest.emailHash);
+    let link = 'https://card-drop-email.test/email-card-drop/verify' + '?' + params.toString();
+
+    await task.perform({ email: 'anyone@test.test', id: cardDropRequest.id }, helpers);
+
+    expect(StubEmail.lastSent.to).to.equal('anyone@test.test');
+    expect(StubEmail.lastSent.from).to.equal(config.get('aws.ses.supportEmail'));
+    expect(StubEmail.lastSent.text).to.contain(link);
+    expect(StubEmail.lastSent.html).to.contain(link);
+  });
+
+  it('fails silently if the row does not exist', async function () {
+    let task = (await getContainer().lookup('send-email-card-drop-verification')) as SendEmailCardDropVerification;
+    let nonexistentId = '6a0d969f-84e1-41d3-8472-87fcf6353476';
+
+    await task.perform({ email: 'anyone@test.test', id: nonexistentId }, helpers);
+
+    expect(StubEmail.lastSent).to.equal(null);
+
+    await waitFor(() => testkit.reports().length > 0);
+
+    let sentryReport = testkit.reports()[0];
+    expect(sentryReport.tags).to.deep.equal({
+      event: 'send-email-card-drop-verification',
+    });
+    expect(sentryReport.error?.message).to.equal(`Unable to find card drop request with id ${nonexistentId}`);
+  });
+
+  it('reports emailing errors to Sentry with the correct tag, then rethrows', async function () {
+    StubEmail.shouldThrow = true;
+
+    let task = (await getContainer().lookup('send-email-card-drop-verification')) as SendEmailCardDropVerification;
+
+    await expect(task.perform({ email: 'anyone@test.test', id: cardDropRequest.id }, helpers)).to.be.rejectedWith(
+      'Intentional error in stub email sending'
+    );
+
+    expect(StubEmail.lastSent).to.equal(null);
+
+    await waitFor(() => testkit.reports().length > 0);
+
+    let sentryReport = testkit.reports()[0];
+    expect(sentryReport.tags).to.deep.equal({
+      event: 'send-email-card-drop-verification',
+    });
```

Comment:
Quite unrelated to this but is there some way we can extract this Sentry-checking, lines like this are now scattered everywhere 😯

Reply:
Unsure, it is pretty common. Probably can make a utility function or custom chai assertion?

Reply:
https://www.chaijs.com/guide/helpers/#adding-methods